airbyte-integrations/connectors/source-google-analytics-v4/source_google_analytics_v4/source.py

```diff
@@ -478,19 +492,17 @@ def check_connection(self, logger, config) -> Tuple[bool, any]:
         config["metrics"] = ["ga:14dayUsers"]
         config["dimensions"] = ["ga:date"]
 
-        # Produce only one date-slice to check the reading permissions
-        first_stream_slice = TestStreamConnection(config).stream_slices(stream_state=None)[0]
-
         try:
             # test the eligibility of custom_reports input
             custom_reports = config.get("custom_reports")
             if custom_reports:
                 json.loads(custom_reports)
 
-            # test the reading operation
-            read_check = list(TestStreamConnection(config).read_records(sync_mode=None, stream_slice=first_stream_slice))
+            # Read records to check the reading permissions
+            read_check = list(TestStreamConnection(config).read_records(sync_mode=None))
             if read_check:
                 return True, None
+            return False, f"Please check the permissions for the requested view_id: {config['view_id']}. No records have been fetched."
```

Comment:
I suggest replacing "No records were fetched" with "Cannot retrieve data from that view ID"